File: CustomerApp/app/src/main/java/com/example/myapplication/model/UserProfile.java
package com.example.myapplication.model;

import java.io.Serializable;

public class UserProfile implements Serializable {

    private String User_name , Email , Mobile , Address ;

    public UserProfile() {
    }

    public UserProfile(String user_name, String email, String mobile, String address) {
        User_name = user_name;
        Email = email;
        Mobile = mobile;
        Address = address;
    }

    public String getUser_name() {
        return User_name;
    }

    public void setUser_name(String user_name) {
        User_name = user_name;
    }

    public String getEmail() {
        return Email;
    }

    public void setEmail(String email) {
        Email = email;
    }

    public String getMobile() {
        return Mobile;
    }

    public void setMobile(String mobile) {
        Mobile = mobile;
    }

    public String getAddress() {
        return Address;
    }

    public void setAddress(String address) {
        Address = address;
    }
}

File: CustomerApp/app/src/main/java/com/example/myapplication/model/UserProfile.kt
package com.example.myapplication.model

import android.os.Parcel
import com.example.myapplication.util.android.KParcelable
import com.example.myapplication.util.android.parcelableCreator
import com.example.myapplication.util.android.readParcelableListCompat
import com.example.myapplication.util.android.writeParcelableListCompat
import kotlinx.serialization.Serializable

private val tempAddresses: MutableList<UserProfile.Address> = mutableListOf()

@Serializable
data class UserProfile(
    val name: String,
    val email: String,
    val phoneNumber: String,
    val addresses: List<Address>
): KParcelable {
    @Serializable
    data class Address(
        val name: String,
        val latitude: Double,
        val longitude: Double,
        val flatNo: String?,
        val building: String,
        val locality: String,
        val city: String,
        val pinCode: String
    ): KParcelable {
        private constructor(parcel: Parcel): this(
            parcel.readString()!!,
            parcel.readDouble(),
            parcel.readDouble(),
            parcel.readString(),
            parcel.readString()!!,
            parcel.readString()!!,
            parcel.readString()!!,
            parcel.readString()!!
        )

        override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
            writeString(name)
            writeDouble(latitude)
            writeDouble(longitude)
            writeString(flatNo)
            writeString(building)
            writeString(locality)
            writeString(city)
            writeString(pinCode)
        }

        companion object {
            @JvmField val CREATOR = parcelableCreator(::Address)
        }
    }

    private constructor(parcel: Parcel): this(
        parcel.readString()!!,
        parcel.readString()!!,
        parcel.readString()!!,
        parcel.readParcelableListCompat(tempAddresses, Address::class.java.classLoader)
    )

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeString(name)
        writeString(email)
        writeString(phoneNumber)
        writeParcelableListCompat(addresses, flags)
    }

    companion object {
        @JvmField val CREATOR = parcelableCreator(::UserProfile)
    }
}